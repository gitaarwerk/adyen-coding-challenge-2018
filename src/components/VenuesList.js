import React from 'react';
import PropTypes from 'proptypes';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemText from '@material-ui/core/ListItemText';
import ListSubheader from '@material-ui/core/ListSubheader';
import Avatar from '@material-ui/core/Avatar';
import CircularProgress from '@material-ui/core/CircularProgress';

const VenuesList = ({ nearbyVenues, isLoading }) => (
  <div>
    {isLoading && (
      <div style={{ color: '#1b2b44', textAlign: 'center', padding: '16px 8px', width: '100%' }}>
        <p>Loading venues...</p>
        <CircularProgress style={{ display: 'inline-block' }} thickness={1} />
      </div>
    )}
    <List
      component="nav"
      subheader={
        <ListSubheader component="div" style={{ fontSize: '2.4rem', color: '#00112c' }}>
          Nearby venues
        </ListSubheader>
      }
    >
      {nearbyVenues.map(item => (
        <ListItem key={item.id} disableGutters={true}>
          <ListItem button>
            <ListItemIcon>
              <Avatar alt={item.icon.name} src={item.icon.url} />
            </ListItemIcon>
            <ListItemText inset primary={item.name} />
          </ListItem>
        </ListItem>
      ))}
    </List>
  </div>
);

VenuesList.propTypes = {
  nearbyVenues: PropTypes.arrayOf(PropTypes.object).isRequired
};

export default VenuesList;
