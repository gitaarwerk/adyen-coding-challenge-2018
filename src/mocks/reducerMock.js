export const geoLocation = {
  latitude: '123',
  longitude: 'abc'
};

export const venues = [
  {
    name: 'Pizza Picasso & Grill',
    address: ['Wagnerplein', '2324 Leiden', 'Netherlands'],
    icon: {
      name: 'Pizza Places',
      url: 'https://ss3.4sqi.net/img/categories_v2/food/pizza_64.png'
    }
  },
  {
    name: 'Adyen',
    address: ['Keizersgracht', 'Amsterdam', 'Netherlands'],
    icon: {
      name: 'Work Places',
      url: 'https://ss3.4sqi.net/img/categories_v2/work/work_64.png'
    }
  }
];

export const fourSquareVenuesResult = [
  {
    id: '4dd569c0d4c05d5096b871d4',
    name: 'Pizza Picasso & Grill',
    location: {
      address: 'Wagnerplein',
      lat: 52.1412748375,
      lng: 4.471563,
      labeledLatLngs: [
        {
          label: 'display',
          lat: 52.1412748375,
          lng: 4.471563
        }
      ],
      distance: 216,
      postalCode: '2324',
      cc: 'NL',
      city: 'Leiden',
      state: 'South Holland',
      country: 'Netherlands',
      formattedAddress: ['Wagnerplein', '2324 Leiden', 'Netherlands']
    },
    categories: [
      {
        id: '4bf58dd8d48988d1ca941735',
        name: 'Pizza Place',
        pluralName: 'Pizza Places',
        shortName: 'Pizza',
        icon: {
          prefix: 'https://ss3.4sqi.net/img/categories_v2/food/pizza_',
          suffix: '.png'
        },
        primary: true
      }
    ],
    referralId: 'v-1533476727',
    hasPerk: false
  }
];
