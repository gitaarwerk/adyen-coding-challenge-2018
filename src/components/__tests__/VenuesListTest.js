import React from 'react';
import { shallow } from 'enzyme';
import { venues } from '../../mocks/reducerMock';

import VenuesList from '../VenuesList';

describe('VenuesList', () => {
  test('renders correctly with venues', () => {
    const component = shallow(<VenuesList isLoading={false} nearbyVenues={venues}/>);

    expect(component).toMatchSnapshot();
  });

  test('renders correctly without venues and loading', () => {
    const component = shallow(<VenuesList isLoading={true} nearbyVenues={[]}/>);

    expect(component).toMatchSnapshot();
  });
});
