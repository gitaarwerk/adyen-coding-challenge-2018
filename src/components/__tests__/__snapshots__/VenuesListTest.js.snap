// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VenuesList renders correctly with venues 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <VenuesList
    isLoading={false}
    nearbyVenues={
        Array [
            Object {
              "address": Array [
                "Wagnerplein",
                "2324 Leiden",
                "Netherlands",
              ],
              "icon": Object {
                "name": "Pizza Places",
                "url": "https://ss3.4sqi.net/img/categories_v2/food/pizza_bg_64.png",
              },
              "id": "4dd569c0d4c05d5096b871d4",
              "name": "Pizza Picasso & Grill",
            },
            Object {
              "address": Array [
                "Keizersgracht",
                "Amsterdam",
                "Netherlands",
              ],
              "icon": Object {
                "name": "Work Places",
                "url": "https://ss3.4sqi.net/img/categories_v2/work/work_bg_64.png",
              },
              "id": "abcd",
              "name": "Adyen",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        false,
        <WithStyles(List)
          component="nav"
          subheader={
                    <WithStyles(ListSubheader)
                              component="div"
                              style={
                                        Object {
                                                  "color": "#00112c",
                                                  "fontSize": "2.4rem",
                                                }
                              }
                    >
                              Nearby venues
                    </WithStyles(ListSubheader)>
          }
>
          <WithStyles(ListItem)
                    disableGutters={true}
          >
                    <WithStyles(ListItem)
                              button={true}
                    >
                              <WithStyles(ListItemIcon)>
                                        <WithStyles(Avatar)
                                                  alt="Pizza Places"
                                                  src="https://ss3.4sqi.net/img/categories_v2/food/pizza_bg_64.png"
                                        />
                              </WithStyles(ListItemIcon)>
                              <WithStyles(ListItemText)
                                        inset={true}
                                        primary="Pizza Picasso & Grill"
                              />
                    </WithStyles(ListItem)>
          </WithStyles(ListItem)>
          <WithStyles(ListItem)
                    disableGutters={true}
          >
                    <WithStyles(ListItem)
                              button={true}
                    >
                              <WithStyles(ListItemIcon)>
                                        <WithStyles(Avatar)
                                                  alt="Work Places"
                                                  src="https://ss3.4sqi.net/img/categories_v2/work/work_bg_64.png"
                                        />
                              </WithStyles(ListItemIcon)>
                              <WithStyles(ListItemText)
                                        inset={true}
                                        primary="Adyen"
                              />
                    </WithStyles(ListItem)>
          </WithStyles(ListItem)>
</WithStyles(List)>,
      ],
    },
    "ref": null,
    "rendered": Array [
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <WithStyles(ListItem)
              disableGutters={true}
>
              <WithStyles(ListItem)
                            button={true}
              >
                            <WithStyles(ListItemIcon)>
                                          <WithStyles(Avatar)
                                                        alt="Pizza Places"
                                                        src="https://ss3.4sqi.net/img/categories_v2/food/pizza_bg_64.png"
                                          />
                            </WithStyles(ListItemIcon)>
                            <WithStyles(ListItemText)
                                          inset={true}
                                          primary="Pizza Picasso & Grill"
                            />
              </WithStyles(ListItem)>
</WithStyles(ListItem)>,
            <WithStyles(ListItem)
              disableGutters={true}
>
              <WithStyles(ListItem)
                            button={true}
              >
                            <WithStyles(ListItemIcon)>
                                          <WithStyles(Avatar)
                                                        alt="Work Places"
                                                        src="https://ss3.4sqi.net/img/categories_v2/work/work_bg_64.png"
                                          />
                            </WithStyles(ListItemIcon)>
                            <WithStyles(ListItemText)
                                          inset={true}
                                          primary="Adyen"
                            />
              </WithStyles(ListItem)>
</WithStyles(ListItem)>,
          ],
          "component": "nav",
          "subheader": <WithStyles(ListSubheader)
            component="div"
            style={
                        Object {
                                    "color": "#00112c",
                                    "fontSize": "2.4rem",
                                  }
            }
>
            Nearby venues
</WithStyles(ListSubheader)>,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "4dd569c0d4c05d5096b871d4",
            "nodeType": "class",
            "props": Object {
              "children": <WithStyles(ListItem)
                button={true}
>
                <WithStyles(ListItemIcon)>
                                <WithStyles(Avatar)
                                                alt="Pizza Places"
                                                src="https://ss3.4sqi.net/img/categories_v2/food/pizza_bg_64.png"
                                />
                </WithStyles(ListItemIcon)>
                <WithStyles(ListItemText)
                                inset={true}
                                primary="Pizza Picasso & Grill"
                />
</WithStyles(ListItem)>,
              "disableGutters": true,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "button": true,
                "children": Array [
                  <WithStyles(ListItemIcon)>
                    <WithStyles(Avatar)
                                        alt="Pizza Places"
                                        src="https://ss3.4sqi.net/img/categories_v2/food/pizza_bg_64.png"
                    />
</WithStyles(ListItemIcon)>,
                  <WithStyles(ListItemText)
                    inset={true}
                    primary="Pizza Picasso & Grill"
/>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <WithStyles(Avatar)
                      alt="Pizza Places"
                      src="https://ss3.4sqi.net/img/categories_v2/food/pizza_bg_64.png"
/>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "alt": "Pizza Places",
                      "src": "https://ss3.4sqi.net/img/categories_v2/food/pizza_bg_64.png",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "inset": true,
                    "primary": "Pizza Picasso & Grill",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "abcd",
            "nodeType": "class",
            "props": Object {
              "children": <WithStyles(ListItem)
                button={true}
>
                <WithStyles(ListItemIcon)>
                                <WithStyles(Avatar)
                                                alt="Work Places"
                                                src="https://ss3.4sqi.net/img/categories_v2/work/work_bg_64.png"
                                />
                </WithStyles(ListItemIcon)>
                <WithStyles(ListItemText)
                                inset={true}
                                primary="Adyen"
                />
</WithStyles(ListItem)>,
              "disableGutters": true,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "button": true,
                "children": Array [
                  <WithStyles(ListItemIcon)>
                    <WithStyles(Avatar)
                                        alt="Work Places"
                                        src="https://ss3.4sqi.net/img/categories_v2/work/work_bg_64.png"
                    />
</WithStyles(ListItemIcon)>,
                  <WithStyles(ListItemText)
                    inset={true}
                    primary="Adyen"
/>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <WithStyles(Avatar)
                      alt="Work Places"
                      src="https://ss3.4sqi.net/img/categories_v2/work/work_bg_64.png"
/>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "alt": "Work Places",
                      "src": "https://ss3.4sqi.net/img/categories_v2/work/work_bg_64.png",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "inset": true,
                    "primary": "Adyen",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          false,
          <WithStyles(List)
            component="nav"
            subheader={
                        <WithStyles(ListSubheader)
                                    component="div"
                                    style={
                                                Object {
                                                            "color": "#00112c",
                                                            "fontSize": "2.4rem",
                                                          }
                                    }
                        >
                                    Nearby venues
                        </WithStyles(ListSubheader)>
            }
>
            <WithStyles(ListItem)
                        disableGutters={true}
            >
                        <WithStyles(ListItem)
                                    button={true}
                        >
                                    <WithStyles(ListItemIcon)>
                                                <WithStyles(Avatar)
                                                            alt="Pizza Places"
                                                            src="https://ss3.4sqi.net/img/categories_v2/food/pizza_bg_64.png"
                                                />
                                    </WithStyles(ListItemIcon)>
                                    <WithStyles(ListItemText)
                                                inset={true}
                                                primary="Pizza Picasso & Grill"
                                    />
                        </WithStyles(ListItem)>
            </WithStyles(ListItem)>
            <WithStyles(ListItem)
                        disableGutters={true}
            >
                        <WithStyles(ListItem)
                                    button={true}
                        >
                                    <WithStyles(ListItemIcon)>
                                                <WithStyles(Avatar)
                                                            alt="Work Places"
                                                            src="https://ss3.4sqi.net/img/categories_v2/work/work_bg_64.png"
                                                />
                                    </WithStyles(ListItemIcon)>
                                    <WithStyles(ListItemText)
                                                inset={true}
                                                primary="Adyen"
                                    />
                        </WithStyles(ListItem)>
            </WithStyles(ListItem)>
</WithStyles(List)>,
        ],
      },
      "ref": null,
      "rendered": Array [
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <WithStyles(ListItem)
                disableGutters={true}
>
                <WithStyles(ListItem)
                                button={true}
                >
                                <WithStyles(ListItemIcon)>
                                                <WithStyles(Avatar)
                                                                alt="Pizza Places"
                                                                src="https://ss3.4sqi.net/img/categories_v2/food/pizza_bg_64.png"
                                                />
                                </WithStyles(ListItemIcon)>
                                <WithStyles(ListItemText)
                                                inset={true}
                                                primary="Pizza Picasso & Grill"
                                />
                </WithStyles(ListItem)>
</WithStyles(ListItem)>,
              <WithStyles(ListItem)
                disableGutters={true}
>
                <WithStyles(ListItem)
                                button={true}
                >
                                <WithStyles(ListItemIcon)>
                                                <WithStyles(Avatar)
                                                                alt="Work Places"
                                                                src="https://ss3.4sqi.net/img/categories_v2/work/work_bg_64.png"
                                                />
                                </WithStyles(ListItemIcon)>
                                <WithStyles(ListItemText)
                                                inset={true}
                                                primary="Adyen"
                                />
                </WithStyles(ListItem)>
</WithStyles(ListItem)>,
            ],
            "component": "nav",
            "subheader": <WithStyles(ListSubheader)
              component="div"
              style={
                            Object {
                                          "color": "#00112c",
                                          "fontSize": "2.4rem",
                                        }
              }
>
              Nearby venues
</WithStyles(ListSubheader)>,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "4dd569c0d4c05d5096b871d4",
              "nodeType": "class",
              "props": Object {
                "children": <WithStyles(ListItem)
                  button={true}
>
                  <WithStyles(ListItemIcon)>
                                    <WithStyles(Avatar)
                                                      alt="Pizza Places"
                                                      src="https://ss3.4sqi.net/img/categories_v2/food/pizza_bg_64.png"
                                    />
                  </WithStyles(ListItemIcon)>
                  <WithStyles(ListItemText)
                                    inset={true}
                                    primary="Pizza Picasso & Grill"
                  />
</WithStyles(ListItem)>,
                "disableGutters": true,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "button": true,
                  "children": Array [
                    <WithStyles(ListItemIcon)>
                      <WithStyles(Avatar)
                                            alt="Pizza Places"
                                            src="https://ss3.4sqi.net/img/categories_v2/food/pizza_bg_64.png"
                      />
</WithStyles(ListItemIcon)>,
                    <WithStyles(ListItemText)
                      inset={true}
                      primary="Pizza Picasso & Grill"
/>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": <WithStyles(Avatar)
                        alt="Pizza Places"
                        src="https://ss3.4sqi.net/img/categories_v2/food/pizza_bg_64.png"
/>,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "alt": "Pizza Places",
                        "src": "https://ss3.4sqi.net/img/categories_v2/food/pizza_bg_64.png",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "inset": true,
                      "primary": "Pizza Picasso & Grill",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "abcd",
              "nodeType": "class",
              "props": Object {
                "children": <WithStyles(ListItem)
                  button={true}
>
                  <WithStyles(ListItemIcon)>
                                    <WithStyles(Avatar)
                                                      alt="Work Places"
                                                      src="https://ss3.4sqi.net/img/categories_v2/work/work_bg_64.png"
                                    />
                  </WithStyles(ListItemIcon)>
                  <WithStyles(ListItemText)
                                    inset={true}
                                    primary="Adyen"
                  />
</WithStyles(ListItem)>,
                "disableGutters": true,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "button": true,
                  "children": Array [
                    <WithStyles(ListItemIcon)>
                      <WithStyles(Avatar)
                                            alt="Work Places"
                                            src="https://ss3.4sqi.net/img/categories_v2/work/work_bg_64.png"
                      />
</WithStyles(ListItemIcon)>,
                    <WithStyles(ListItemText)
                      inset={true}
                      primary="Adyen"
/>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": <WithStyles(Avatar)
                        alt="Work Places"
                        src="https://ss3.4sqi.net/img/categories_v2/work/work_bg_64.png"
/>,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "alt": "Work Places",
                        "src": "https://ss3.4sqi.net/img/categories_v2/work/work_bg_64.png",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "inset": true,
                      "primary": "Adyen",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`VenuesList renders correctly without venues and loading 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <VenuesList
    isLoading={true}
    nearbyVenues={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          style={
                    Object {
                              "color": "#1b2b44",
                              "padding": "16px 8px",
                              "textAlign": "center",
                              "width": "100%",
                            }
          }
>
          <p>
                    Loading venues...
          </p>
          <WithStyles(CircularProgress)
                    style={
                              Object {
                                        "display": "inline-block",
                                      }
                    }
                    thickness={1}
          />
</div>,
        <WithStyles(List)
          component="nav"
          subheader={
                    <WithStyles(ListSubheader)
                              component="div"
                              style={
                                        Object {
                                                  "color": "#00112c",
                                                  "fontSize": "2.4rem",
                                                }
                              }
                    >
                              Nearby venues
                    </WithStyles(ListSubheader)>
          }
>
          
</WithStyles(List)>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p>
              Loading venues...
</p>,
            <WithStyles(CircularProgress)
              style={
                            Object {
                                          "display": "inline-block",
                                        }
              }
              thickness={1}
/>,
          ],
          "style": Object {
            "color": "#1b2b44",
            "padding": "16px 8px",
            "textAlign": "center",
            "width": "100%",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Loading venues...",
            },
            "ref": null,
            "rendered": "Loading venues...",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "style": Object {
                "display": "inline-block",
              },
              "thickness": 1,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [],
          "component": "nav",
          "subheader": <WithStyles(ListSubheader)
            component="div"
            style={
                        Object {
                                    "color": "#00112c",
                                    "fontSize": "2.4rem",
                                  }
            }
>
            Nearby venues
</WithStyles(ListSubheader)>,
        },
        "ref": null,
        "rendered": Array [],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            style={
                        Object {
                                    "color": "#1b2b44",
                                    "padding": "16px 8px",
                                    "textAlign": "center",
                                    "width": "100%",
                                  }
            }
>
            <p>
                        Loading venues...
            </p>
            <WithStyles(CircularProgress)
                        style={
                                    Object {
                                                "display": "inline-block",
                                              }
                        }
                        thickness={1}
            />
</div>,
          <WithStyles(List)
            component="nav"
            subheader={
                        <WithStyles(ListSubheader)
                                    component="div"
                                    style={
                                                Object {
                                                            "color": "#00112c",
                                                            "fontSize": "2.4rem",
                                                          }
                                    }
                        >
                                    Nearby venues
                        </WithStyles(ListSubheader)>
            }
>
            
</WithStyles(List)>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p>
                Loading venues...
</p>,
              <WithStyles(CircularProgress)
                style={
                                Object {
                                                "display": "inline-block",
                                              }
                }
                thickness={1}
/>,
            ],
            "style": Object {
              "color": "#1b2b44",
              "padding": "16px 8px",
              "textAlign": "center",
              "width": "100%",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Loading venues...",
              },
              "ref": null,
              "rendered": "Loading venues...",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "style": Object {
                  "display": "inline-block",
                },
                "thickness": 1,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [],
            "component": "nav",
            "subheader": <WithStyles(ListSubheader)
              component="div"
              style={
                            Object {
                                          "color": "#00112c",
                                          "fontSize": "2.4rem",
                                        }
              }
>
              Nearby venues
</WithStyles(ListSubheader)>,
          },
          "ref": null,
          "rendered": Array [],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
